wordpress-mobile/WordPress-Android
# Magic link is the new login experience

Fixes #4461 

We are going with magic links as the primary way to log in (at least for WordPress.com)

Because `MagicLinkSignInActivity/Fragment` was simply subclassing/overriding `SignInActivity/Fragment` it's fairly straight forward. All overridden/new methods have been added.

To test:
Go through all log-in situations
1. Log in with a username/password
2. Log in with an email address/magic link
3. Log in with an email address/password
4. Verify smart lock auto-fills account details if the user already has a password saved
5. Log into a self-hosted site
6. Create a new account on WordPress.com

Needs review: @maxme
